### PR TITLE
build: always report the status of building `_Differentiation`

### PR DESCRIPTION
**Pull request by compnerd**
This moves the build configuration information into the block which
prints the full configuration details for the Swift runtime.  Simplify
the condition by merging the two blocks for building the standard
library and the `_Differentiation` module.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
CC: @dan-zheng

File: CMakeLists.txt [hunk 1]
```diff
@@ -876,6 +876,7 @@ if(SWIFT_BUILD_STDLIB OR SWIFT_BUILD_SDK_OVERLAY)
 
   message(STATUS "Building Swift runtime with:")
   message(STATUS "  Leak Detection Checker Entrypoints: ${SWIFT_RUNTIME_ENABLE_LEAK_CHECKER}")
+  message(STATUS "  Differentiable Programming Support: ${SWIFT_ENABLE_EXPERIMENTAL_DIFFERENTIABLE_PROGRAMMING}")
```
Comment (dan-zheng) [inline CMakeLists.txt]: Isn't it misleading to put this message under `Building Swift runtime with:`? How about moving the message somewhere else?

Differentiable programming doesn't currently involve Swift runtime changes. I think of `_Differentiation` as more like an auxiliary library, similar to SDK overlays.

**Review by dan-zheng — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (compnerd) [reply CMakeLists.txt]: Ah, I see what you mean.  Sure, I'll outdent it.

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Review by dan-zheng — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Review by rxwei — APPROVED** [review]

**Comment by rxwei** [comment]
Minor correction: `Differentiation` module -> `_Differentiation` module

**Review by Rostepher — APPROVED** [review]
LGTM!

**Pull request merged**